airflow/models/dagbag.py

```diff
@@ -512,6 +514,16 @@ def _bag_dag(self, *, dag, root_dag, recursive):
             settings.dag_policy(dag)
 
             for task in dag.tasks:
+                # The listeners are not supported when ending a task via a trigger on asynchronous operators.
+                if getattr(task, "end_from_trigger", False) and get_listener_manager().has_listeners:
+                    raise AirflowException(
+                        "Listeners are not supported with end_from_trigger=True for async operators. "
```

Comment:
```suggestion
                        "Listeners are not supported with end_from_trigger=True for deferrable operators. "
```

not sure whether "deferrable operators" is the best choice, but the term "async op" doesn't seem to be used that often in airflow